lib/blocktree/node.go

```diff
@@ -55,17 +55,24 @@ func (n *node) createTree(tree gotree.Tree) {
 
 // getNode recursively searches for a node with a given hash
 func (n *node) getNode(h common.Hash) *node {
+	if n == nil {
+		return nil
+	}
```

Comment:
nit Does this happen? Should we check at the caller level if node is nil or not instead of calling `getNode` on it perhaps?

Reply:
it shouldn't happen, but I'd rather have this check here just to be safe... do you know of a good method for dealing with things like this, where it *shouldn't* happen but if it does it's really bad? kinda what I thought `panic`s were for, but seems it's not good to use those

Reply:
In my opinion, panics are when you can't break a function signature. For example if you have an exported, public `Divide(a, b int) int` and later realise you cannot divide with `b = 0`, use a `panic`.

I usually try to return an error instead of panicing. In this situation though, I think you are returning a `nil` result. If this is not case meant to be handled there, you should either return an error or panic there. If it's really critical and that would mess up everything else, then panic for sure.

The best long term solution is testing depth and coverage, but that's not an easy task!